test/e2e/examples.go

```diff
@@ -0,0 +1,207 @@
+/*
+Copyright 2015 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package e2e
+
+import (
+	"fmt"
+	"path/filepath"
+	"strings"
+	"time"
+
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/api"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/client"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/fields"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/labels"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+)
+
+var _ = Describe("Examples e2e", func() {
+	defer GinkgoRecover()
+	var c *client.Client
+	var ns string
+	var testingNs *api.Namespace
+	BeforeEach(func() {
+		var err error
+		c, err = loadClient()
+		expectNoError(err)
+		testingNs, err = createTestingNS("examples", c)
+		ns = testingNs.Name
+		Expect(err).NotTo(HaveOccurred())
+	})
+
+	AfterEach(func() {
+		By(fmt.Sprintf("Destroying namespace for this suite %v", ns))
+		if err := c.Namespaces().Delete(ns); err != nil {
+			Failf("Couldn't delete ns %s", err)
+		}
+	})
+
+	Describe("Redis example", func() {
+		It("should create and stop redis servers", func() {
+			mkpath := func(file string) string {
+				return filepath.Join(testContext.RepoRoot, "examples/redis", file)
+			}
+			bootstrapYaml := mkpath("redis-master.yaml")
+			sentinelServiceYaml := mkpath("redis-sentinel-service.yaml")
+			sentinelControllerYaml := mkpath("redis-sentinel-controller.yaml")
+			controllerYaml := mkpath("redis-controller.yaml")
+
+			bootstrapPodName := "redis-master"
+			redisRC := "redis"
+			sentinelRC := "redis-sentinel"
+			nsFlag := fmt.Sprintf("--namespace=%v", ns)
+
+			expectedOnServer := "The server is now ready to accept connections"
+			expectedOnSentinel := "+monitor master"
+			startupTimeout := 3 * time.Minute
+
+			cleanupBootstrap := true
+			defer func() {
+				if cleanupBootstrap {
+					cleanup(bootstrapYaml, ns, "name=redis", "name=redis-sentinel")
+				}
+			}()
+
+			defer cleanup(sentinelServiceYaml, ns, "name=redis-sentinel")
+			defer cleanup(sentinelControllerYaml, ns, "name=redis-sentinel")
+			defer cleanup(controllerYaml, ns, "name=redis")
+
+			By("starting redis bootstrap")
+			runKubectl("create", "-f", bootstrapYaml, nsFlag)
+			_, err := lookForStringInLog(ns, bootstrapPodName, "master", expectedOnServer, startupTimeout)
```

Comment:
It checks that the master redis server is fully up and running, not just that the pod is running. Nevertheless I will add waitForPodRunningInNamespace here as the pod may still be in the pending state when kubectl is invoked.